pkg/gardenlet/controller/managedseed/managedseed.go

```diff
@@ -119,20 +119,23 @@ func (c *Controller) Run(ctx context.Context, workers int) {
 	})
 
 	// Add event handler for controlled seeds
-	c.seedInformer.AddEventHandler(&kutils.ControlledResourceEventHandler{
-		ControllerTypes: []kutils.ControllerType{
-			{
-				Type:      &seedmanagementv1alpha1.ManagedSeed{},
-				Namespace: pointer.StringPtr(gardencorev1beta1constants.GardenNamespace),
-				NameFunc:  func(obj client.Object) string { return obj.GetName() },
+	c.seedInformer.AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: controllerutils.SeedOfManagedSeedFilterFunc(ctx, c.gardenClient.Client(), confighelper.SeedNameFromSeedConfig(c.config.SeedConfig), c.config.SeedSelector),
```

Comment:
Passing the cache down as a reader is probably better in the event handler. It will ensure, that we don't read from the API server again even though we just have received a watch event for `ManagedSeed`/`Seed`, in case the `CachedRuntimeClients` feature gate is not enabled.
Could be done for the `managedSeedInformer` as well (just noticed it...).

```suggestion
		FilterFunc: controllerutils.SeedOfManagedSeedFilterFunc(ctx, c.gardenClient.Cache(), confighelper.SeedNameFromSeedConfig(c.config.SeedConfig), c.config.SeedSelector),
```

Reply:
Done as suggested